### DISCUSSION TITLE
FastDifferentiatoin as Derivative Backend for Symbolics

### DISCUSSION THREAD

**Discussion by ejmeitz** [brianguenter/FastDifferentiation.jl#58] [Ideas]
What work would need to be done for `Symbolics.jl` to use `FastDifferentiatoin.jl` as its differentiation back-end? I noticed that the derivative in Symbolics are relatively slow and you can also not take derivatives of expressions defined in terms of array variables. There are a lot of similarlites between the two packages and it would be cool if Symbolics could rely on FastDiff.

**Comment by brianguenter** [comment]
Yeah, I've been talking to them about this for quite a while. The interface
code is not too bad but I've been tracking down bugs in FastDifferentiation
in the, perhaps mistaken, belief that everything has to be perfect before
incorporating it into Symbolics.jl.

I'm also working on new code to do very efficient higher order derivatives.

I should push this through. Just need to find time to finish it up.
